src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -8572,11 +8572,37 @@ def wrapped_func(*args, **kwargs):  # type: ignore[no-untyped-def] # pragma: no
     def _apply_snowpark_python_function_to_columns(
         self,
         snowpark_function: Callable,
+        kwargs: Any,  # possible named arguments which need to be added
     ) -> "SnowflakeQueryCompiler":
         """Apply Snowpark Python function to columns."""
 
         def sf_function(col: SnowparkColumn) -> SnowparkColumn:
-            return snowpark_function(col)
+            if not kwargs:
+                return snowpark_function(col)
+            # we have named kwargs, which may be positional
+            # in nature, and we need to align them to the snowpark
+            # function call alongside the column reference
+            # Get the total arg count for the function
+            function_arg_count = snowpark_function.__code__.co_argcount
+            # Get all variables for the function and slice off only the arguments
+            positional_args = snowpark_function.__code__.co_varnames[
+                :function_arg_count
+            ]
+            resolved_positional = []
+            col_specified = False
+            for arg in positional_args:
+                if arg in kwargs:
+                    resolved_positional.append(kwargs[arg])
+                else:
+                    if not col_specified:
+                        resolved_positional.append(col)
+                        col_specified = True
+                    else:
+                        ErrorMessage.not_implemented(
+                            "kwargs specified with apply cannot have multiple column references"
```

Comment:
can you provide more info like which key has multiple column references?

Reply:
Ok. I re-worded things and added a test to make sure I was catching this case. The error message is a little complicated, but I summed it up to something more like:

"Unspecified Argument: {arg} - when using apply with kwargs, all function arguments should be specified except the single column reference (if applicable)."

It's a bit of a weird scenario, but if we are limiting this to snowpark functions for now I'm not sure if the user would run into this accidentally.

Reply:
I just want to deliver more info from this message back to our telemetry so we can reason and reproduce on our side.